### ISSUE TITLE
office-addin-lint-check with syntax error on shell script after 1.1.4

### ISSUE THREAD

**Issue by villasv** [OfficeDev/Office-Addin-Scripts#410]
# Expected behavior

`npm run lint` to run without errors on both Windows and WSL.

# Current behavior

Exit 0 on Windows.
Exit 2 on Ubuntu (WSL):

```
npm run lint

> home-analyst-excel-addin@0.0.1 lint /home/villasv/projects/HomeAnalyst
> office-addin-lint check

/bin/sh: 1: Syntax error: "(" unexpected
```

## Steps to Reproduce

Should be reproducible in any project by running `office-addin-lint check` inside a POSIX shell.

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* Operating System: Ubuntu 20.04.2 LTS
* Node version: v12.18.4
* Office version:
* Tool version: **office-addin-lint 1.1.4** (version 1.1.3 worked fine)

**Comment by villasv** [comment]
I've searched for the commits between 1.1.13 and 1.1.14, but the only parenthesis related code change I could find was this:

https://github.com/OfficeDev/Office-Addin-Scripts/pull/406/files#diff-181137f1a67947eeb0e58a3bc1f8e03141fbf90276934118dc1598957735b1cfR6

Seemed benign, but I can confirm that changing only that `"src/**/*.{ts?(x),js?(x)}"` into `"src/**/*.{ts,tsx,js,jsx}"` directly inside my `node_modules/office-addin-lint/lib/defaults.js` file, the error goes away.

**Comment by villasv** [comment]
Before: 
```
sh -c "node /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/node_modules/eslint/bin/eslint.js -c /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/config/.eslintrc.json --resolve-plugins-relative-to /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/lib src/**/*.{ts,tsx,js,jsx}"
```
After:
```
$ sh -c "node /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/node_modules/eslint/bin/eslint.js -c /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/config/.eslintrc.json --resolve-plugins-relative-to /home/villasv/projects/HomeAnalyst/node_modules/office-addin-lint/lib src/**/*.{ts?(x),js?(x)}"
sh: 1: Syntax error: "(" unexpected
```

**Comment by akrantz** [comment]
You can override this in your package.json by setting  config: { "lint-files": "src/**/*.{ts,tsx,js,jsx}` }

**Comment by akrantz** [comment]
The previous lint-files value would generate an error if no *.tsx files were present. The change fixed that although it appears to not work on WSL.  Do you know a way to make this work on all platforms? Simply reverting this isn't really useful as it just restores the other problem.  The workaround I provided allows you to override this in your own project. Does this work for you?

**Comment by villasv** [comment]
> Does this work for you?

It works in the sense that it stops the error, but I'm interested in trying to find a cross-platform default. I wasn't aware the previous default had it's own issue, my bad for not reading the commit message: `allow lint command to work when there are no tsx/jsx files`.

Was that error specific to Windows? Because I don't have js or jsx and it's not giving me errors on neither OS.

**Comment by villasv** [comment]
> The previous lint-files value would generate an error if no *.tsx files were present. 

Testing locally, both on Windows and Linux it would only error if there were no ts, tsx, js, jsx at all for me. Having a single one of those it would work.

```
Oops! Something went wrong! :(

ESLint: 7.22.0

No files matching the pattern "src/**/*.{ts,tsx,js,jsx}" were found.
Please check for typing mistakes in the pattern.
```

Is there a reproducible example for me to test the problem solved by commit 3fcac177eefd25218ccff01e750b69eb4ab600ea?

**Comment by akrantz** [comment]
Well I am perplexed. 

If I run `npm run lint -- --files src/**/*.tsx`, then it will give an error.

```
Oops! Something went wrong! :(

ESLint: 7.22.0

No files matching the pattern "src/**/*.tsx" were found.
Please check for typing mistakes in the pattern.
```

But if I run: `npm run lint -- --files src/**/*.{ts,tsx,js,jsx}` it will succeed now on Windows.

This is using the latest packages from https://github.com/OfficeDev/Office-Addin-Scripts (office-addin-lint, eslint-config-office-addins, office-addins-prettier-config).

I have WSL with Ubuntu 18.04, and I can verify the error. Using `npm run lint --  --files src/**/*.{ts,tsx,js,jsx}` will succeed.

I think you're right that this was a bad change and we should revert it. You can re-open your PR. Thanks.
